tests/test_questionnaire_validator.py

```diff
@@ -408,6 +412,66 @@ def test_invalid_calculated_or_grand_calculated_summary_id_in_value_source():
     assert validator.errors == expected_errors
 
 
+def test_answer_as_source_referenced_before_created():
+    filename = "schemas/invalid/test_invalid_answer_source_reference.json"
+    validator = QuestionnaireValidator(_open_and_load_schema_file(filename))
+
+    expected_errors = [
+        {
+            "block_id": "confirm-dob",
+            "message": "Answer 'date-of-birth-answer' referenced as source before it has "
+            "been added.",
+        },
+        {
+            "block_id": "confirm-dob",
+            "message": "Answer 'date-of-birth-answer' referenced as source before it has "
+            "been added.",
+        },
+        {
+            "block_id": "confirm-dob",
+            "message": "Answer 'date-of-birth-answer' referenced as source before it has "
+            "been added.",
+        },
```

Comment:
Only a minor comment, and up to you wether you want to change this, but from the surface these look like duplicates, but it is only when you check the test schema you can see that these are valid. Potentially could have added a couple of other blocks to be referenced as answers so that the id's are different, but only a minor thing.

Reply:
I have updated the schema to add a new `confirm-sex` block and removed some of the placeholders  `date-of-birth-answer `in the `confirm-dob` block to reduce the duplication.